runtime/src/block_component_processor.rs

```diff
@@ -133,6 +137,52 @@ impl BlockComponentProcessor {
         Ok(())
     }
 
+    fn enforce_alpenglow_clock_bounds(
+        bank: Arc<Bank>,
+        parent_bank: Arc<Bank>,
+        footer: &BlockFooterV1,
```

Comment:
dunno how you handle BlockFooter versioning, but I sort of expected BlockFooterCurrent here and footer.block_producer_time_nanos would work in later vesions.